diff --git a/index.js b/index.js
index f54765d7536bc51b76728e2a375f79ff09f814fc..9ae0b98addaba7abc6f26bc6fedd97ac06b8e05e 100644
--- a/index.js
+++ b/index.js
@@ -661,7 +661,13 @@ async function create_function_bundle(builder, entry, dir, config) {
 	let base = entry;
 	while (base !== (base = path.dirname(base)));
 
-	const traced = await nodeFileTrace([entry], { base });
+	const is_vercel_build = !!process.env.VERCEL && entry.startsWith('/vercel/path0');
+
+	const traced = await nodeFileTrace([entry], {
+		base,
+		// TODO: remove this when https://github.com/vercel/nft/pull/568 is resolved
+		ignore: is_vercel_build ? (path) => !path.match(/^\/?vercel\/path0/) : undefined
+	});
 
 	/** @type {Map<string, string[]>} */
 	const resolution_failures = new Map();
