diff --git a/out/analyze.js b/out/analyze.js
index 948bb0dec5453dfffa3eebcfe7ee1080c7b2071a..28bf43e3746df0c59c3092f3aad9d5f7e5569794 100644
--- a/out/analyze.js
+++ b/out/analyze.js
@@ -994,9 +994,52 @@ async function analyze(id, code, job) {
         if (wildcardIndex !== -1 && stats.isFile())
             return;
         if (stats.isFile()) {
+                  // do not emit file assets outside job.base
+      if (job.ignoreFn(path_1.relative(job.base, assetPath))) return;
+      // do not emit file assets outside the package boundary if inside node_modules
+      if (pkgBase) {
+        const nodeModulesBase =
+          id.substring(0, id.indexOf(path_1.sep + 'node_modules')) +
+          path_1.sep +
+          'node_modules' +
+          path_1.sep;
+        if (!assetPath.startsWith(nodeModulesBase)) {
+          if (job.log)
+            console.log(
+              'Skipping asset emission of ' +
+                assetPath +
+                ' for ' +
+                id +
+                ' as it is outside the package base ' +
+                pkgBase,
+            );
+          return;
+        }
+      }
             assets.add(assetPath);
         }
         else if (stats.isDirectory()) {
+                  // do not emit directory assets outside job.base
+      if (job.ignoreFn(path_1.relative(job.base, assetPath))) return;
+      if (pkgBase) {
+        const nodeModulesBase =
+          id.substring(0, id.indexOf(path_1.sep + 'node_modules')) +
+          path_1.sep +
+          'node_modules' +
+          path_1.sep;
+        if (!assetPath.startsWith(nodeModulesBase)) {
+          if (job.log)
+            console.log(
+              'Skipping asset emission of ' +
+                assetPath +
+                ' for ' +
+                id +
+                ' as it is outside the package base ' +
+                pkgBase,
+            );
+          return;
+        }
+      }
             if (validWildcard(assetPath))
                 emitAssetDirectory(assetPath);
         }
diff --git a/out/utils/ast-helpers.js b/out/utils/ast-helpers.js
index 72479bdd04d4f27b4cac53d87ce0c588bb1b8df8..0b508494c04affb6cfa0b30e6944bbc2c55888f4 100644
--- a/out/utils/ast-helpers.js
+++ b/out/utils/ast-helpers.js
@@ -25,7 +25,7 @@ function isIdentifierRead(node, parent) {
             return parent.id !== node;
         // disregard the `bar` in `export { foo as bar }`
         case 'ExportSpecifier':
-            return false;
+            return node.name === parent.exported.name;
         // disregard the `bar` in `function (bar) {}`
         case 'FunctionExpression':
         case 'FunctionDeclaration':
